hera_cal/datacontainer.py

```diff
@@ -433,3 +433,48 @@ def has_pol(self, pol):
     def get(self, antpair, pol):
         '''Interface to DataContainer.__getitem__(bl + (pol,)).'''
         return self[make_bl(antpair, pol)]
+
+    def select_or_expand_times(self, new_times, in_place=True):
+        '''Update self.times with new times, updating data and metadata to be consistent. Data and
+        metadata will be deleted, rearranged, or duplicated as necessary using numpy's fancy indexing.
+        Assumes that the 0th data axis is time. Does not support baseline-dependent averaging.
+
+        Arguments:
+            new_times: list or numpy array of times to use to index into this object. These must all be in
+                self.times, but they can be a subset in any order with any number of duplicates.
+            in_place: if True, this DataContainer is modified. Otherwise, a modified copy is returned.
+        '''
+        if in_place:
+            dc = self
+        else:
+            dc = copy.deepcopy(self)
+
+        # make sure this is a sensible object for performing this operation
+        assert dc.times is not None
+        assert np.all([nt in dc.times for nt in new_times]), 'All new_times must be in self.times.'
```

Comment:
Same comment as above about bare `assert` statements.

Reply:
I've done this for the one where the input is bad. The other circumstances where the object is a very non-standard data container I think are rare enough that we're talking about a power user anyway.